dataset_analysis/generate_dataframe.py

```diff
@@ -0,0 +1,462 @@
+"""
+This file generates the dataframe used for the analysis of the CanProCo dataset.
+For each patients in the CanProCo dataset, it gathers information about the patient, the pathology and the lesion(s).
+
+
+Args:
+    --data : path to the CanProCo dataset
+    --lesion : path to the lesion segmentation file
+    --discs : path to the vertebral levels segmentation file
+    --spinal-cord : path to the spinal cord segmentation file
+    --timepoint : timepoint of the analysis (M0, M12)
+    --exclude-file : a .yml file containing the list of participants to exclude
+    --output : path to the output folder
+
+Returns:
+    None
+
+Example:
+    python generate_dataframe.py --data /path/to/CanProCo --lesion /path/to/lesion/segmentation --discs /path/to/discs/segmentation --spinal-cord /path/to/spinal/cord/segmentation --timepoint M0 --exclude-file /path/to/exclude/file --output /path/to/output/folder
+
+Todo:
+    *
+
+Pierre-Louis Benveniste
+"""
+
+import os
+import argparse
+from pathlib import Path
+import pandas as pd
+import json
+import shutil
+import nibabel as nib
+import numpy as np
+import yaml
+from image import Image, get_dimension, change_orientation
+
+
+def get_parser():
+    """
+    This function parses the command line arguments and returns an argparse object.
+
+    Input:
+        None
+
+    Returns:
+        parser : argparse object
+    """
+    parser = argparse.ArgumentParser(description='Generates the dataset used for the analysis of the CanProCo dataset.')
+    parser.add_argument('--data', '-d', type=str, help='path to the CanProCo dataset')
+    parser.add_argument('--lesion', '-l', type=str, help='path to the lesion segmentation file')
+    parser.add_argument('--discs', type=str, help='path to the vertebral levels segmentation file')
+    parser.add_argument('--spinal-cord', type=str, help='path to the spinal cord segmentation file')
+    parser.add_argument('--timepoint', '-t', type=str, help='timepoint of the analysis (M0, M12)')
+    parser.add_argument('--exclude-file', '-e', type=str, help='a .yml file containing the list of participants to exclude')
+    parser.add_argument('--output', '-o', type=str, help='path to the output folder')
+
+    return parser
+
+
+def get_spinal_cord_info(patient_data, spinal_cord_path, timepoint):
```

Comment:
A better name would be `compute_spinal_cord_volume`

Reply:
Is the SC mask cropped to certain vert levels before computing the volume? Such a step should be done because otherwise the volume might differ based on the FOV.

Reply:
I didn't change the name because in the future we might add other information about the spinal cord, like the excentricity...

Reply:
Nothing is cropped for now. Certain levels could be selected to have something more robust to different FOV. Which cropping would you do ?

Reply:
> Nothing is cropped for now. Certain levels could be selected to have something more robust to different FOV. 

Well, then the volume is really only informative.

> Which cropping would you do ?

Depends on the area we are interested in. If cervical, then C1-C7.

Reply:
On it !

Reply:
Done !